modules/ztp-image-based-upgrade-concept.adoc

```diff
@@ -0,0 +1,133 @@
+plan:
+- actions: ["Prep", "Upgrade", "FinalizeUpgrade"]
+  rolloutStrategy:
+    maxConcurrency: 200
+    timeout: 60
+----
+a|[source,yaml]
+----
+plan:
+- actions: ["Prep", "Upgrade"]
+  rolloutStrategy:
+    maxConcurrency: 200
+    timeout: 60
+- actions: ["FinalizeUpgrade"]
+  rolloutStrategy:
+    maxConcurrency: 500
+    timeout: 10
+----
+a|[source,yaml]
+----
+plan:
+- actions: ["Prep"]
+  rolloutStrategy:
+    maxConcurrency: 200
+    timeout: 60
+- actions: ["Upgrade"]
+  rolloutStrategy:
+    maxConcurrency: 200
+    timeout: 20
+- actions: ["FinalizeUpgrade"]
+  rolloutStrategy:
+    maxConcurrency: 500
+    timeout: 10
+----
+|====
+
+[IMPORTANT]
+====
+Clusters that fail one of the actions will skip the remaining actions in the same step.
+====
+
+The `ImageBasedGroupUpgrade` API accepts the following actions:
+
+`Prep`:: Start preparing the upgrade resources by moving to the `Prep` stage.
+`Upgrade`:: Start the upgrade by moving to the `Upgrade` stage.
+`FinalizeUpgrade`:: Finalize the upgrade on selected clusters that completed the `Upgrade` action by moving to the `Idle` stage.
+`Rollback`:: Start a rollback only on successfully upgraded clusters by moving to the `Rollback` stage.
+`FinalizeRollback`:: Finalize the rollback by moving to the `Idle` stage.
+`AbortOnFailure`:: Cancel the upgrade on selected clusters that failed the `Prep` or `Upgrade` actions by moving to the `Idle` stage.
+`Abort`:: Cancel an ongoing upgrade only on clusters that are not yet upgraded by moving to the `Idle` stage.
+
+The following action combinations are supported. A pair of brackets signifies one step in the `plan` section:
+
+* `["Prep"]`, `["Abort"]`
+* `["Prep", "Upgrade", "FinalizeUpgrade"]`
+* `["Prep"]`, `["AbortOnFailure"]`, `["Upgrade"]`, `["AbortOnFailure"]`, `["FinalizeUpgrade"]`
+* `["Rollback", "FinalizeRollback"]`
+
+Use one of the following combinations when you need to resume or cancel an ongoing upgrade from a completely new `ImageBasedGroupUpgrade` CR:
+
+* `["Upgrade","FinalizeUpgrade"]`
+* `["FinalizeUpgrade"]`
+* `["FinalizeRollback"]`
+* `["Abort"]`
+* `["AbortOnFailure"]`
+
+[id="ztp-image-based-upgrade-cluster-labeling_{context}"]
+== Labeling for cluster selection
+
+Use the `clusterLabelSelectors` field for initial cluster selection.
+{cgu-operator} labels the managed clusters according to the results of their last stage transition.
+
+When a stage completes or fails, {cgu-operator} marks the relevant clusters with the following labels:
+
+* `lcm.openshift.io/ibgu-<stage>-completed`
+* `lcm.openshift.io/ibgu-<stage>-failed`
+
+For example, if you want to cancel the upgrade for all managed clusters except for clusters that successfully completed the upgrade, you can add an `Abort` action to your plan.
+This ensures that all clusters that have the `lcm.openshift.io/ibgu-upgrade-completed` label are not moved back to the `Idle` stage.
+
+Transition labels are removed after successfully canceling or finalizing the change.
```

Comment:
```suggestion
These cluster labels are removed after executing the 'Abort' action plan successfully which performs canceling or finalizing the change.
```

Reply:
This implies that the `Abort` action cancels _and_ finalizes the change, which while technically true, this can be confusing. We've been using these two phrases for different scenarios:
- Canceling: move back to Idle before the Upgrade stage could finish. This is Abort but as per the conscious language initiative, we can't use abort, instead we use cancel.
- Finalize change: Move to Idle after a successful Upgrade or Rollback

Reply:
Okay. In my view, Transition labels is uncertain from end-users perspective - I tend to question myself `what is Transition labels` and for the rest - I will leave it to you to make a final conclusion.

Reply:
I changed this to cluster labels to avoid confusion but I kept the rest of the sentence. So the current version is:
> The cluster labels are removed after successfully canceling or finalizing the change.

Reply:
Okay, Looks good to me.